pkg/gui/controllers/helpers/app_status_helper.go

```diff
@@ -73,3 +85,31 @@ func (self *AppStatusHelper) renderAppStatus() {
 		}
 	})
 }
+
+func (self *AppStatusHelper) renderAppStatusSync(stop chan struct{}) {
+	ticker := time.NewTicker(time.Millisecond * 50)
+	go func() {
+		// Forcing a re-layout and redraw after we added the waiting status;
+		// this is needed in case the gui.showBottomLine config is set to false,
+		// to make sure the bottom line appears. It's also useful for redrawing
+		// once after each of several consecutive keypresses, e.g. pressing
+		// ctrl-j to move a commit down several steps.
+		_ = self.c.GocuiGui().ForceLayoutAndRedraw()
+
+		self.c.SetFreezeInformationView(true)
+		defer func() { self.c.SetFreezeInformationView(false) }()
+
+	outer:
+		for {
+			select {
+			case <-ticker.C:
+				appStatus := self.statusMgr().GetStatusString()
+				self.c.SetViewContent(self.c.Views().AppStatus, " "+appStatus)
+				_ = self.c.GocuiGui().ForceRedrawViews(self.c.Views().AppStatus, self.c.Views().Options)
+			case <-stop:
+				break outer
+			}
+		}
+		ticker.Stop()
```

Comment:
I'd define the ticker inside the goroutine and then use `defer ticker.Stop()` right after so that there's no risk of issues from early returns

Reply:
Fixed in eb9f3a5d.